### DISCUSSION TITLE
Cannot find "usbCamera" plugin while projecting streaming to yarp port?

### DISCUSSION THREAD

**Discussion by YANGGANG1971** [robotology/community#409] [Q&A] [answered]
Following the [guidance ](https://github.com/robotology/QA/issues/402), the integrated webcam can be projected to iCub simulator successfully. Thanks for your help. 

Recently, I am trying to use the desktop to do some experiments. There is no integrated webcam. So I moved on to using the usb camera by video4linux. 

1. ccmake ../    (recomplie yarp)
2. ENABLE_yarpmod_grabber ON
3. make 
4. yarpdev --device grabber --subdevice usbCamera --d /dev/video2
 (The external usbcam streaming is /dev/video2.)
But, each time, Cannot find "usbCamera" plugin will occur as follows. 
> [ERROR]/usr/local/src/robot/yarp/src/libYARP_os/src/yarp/os/YarpPluginSettings.h:240 bool yarp::os::YarpPluginSettings::readFromSelector(const string&): Cannot find "usbCamera" plugin (not built in, and no .ini file found for it)
> Check that YARP_DATA_DIRS leads to at least one directory with plugins/usbCamera.ini or share/yarp/plugins/usbCamera.ini in it
> yarpdev: ***ERROR*** could not find device <usbCamera>
> yarpdev: ***ERROR*** driver <grabber> was found but could not open
> [ERROR]/usr/local/src/robot/yarp/src/libYARP_dev/src/yarp/dev/Drivers.cpp:503 static int yarp::dev::Drivers::yarpdev(int, char**): yarpdev: ***ERROR*** device not available.
> [INFO]/usr/local/src/robot/yarp/src/libYARP_dev/src/yarp/dev/Drivers.cpp:511 static int yarp::dev::Drivers::yarpdev(int, char**): Suggestions:
> [INFO]/usr/local/src/robot/yarp/src/libYARP_dev/src/yarp/dev/Drivers.cpp:512 static int yarp::dev::Drivers::yarpdev(int, char**): + Do "yarpdev --list" to see list of supported devices.
> [INFO]/usr/local/src/robot/yarp/src/libYARP_dev/src/yarp/dev/Drivers.cpp:515 static int yarp::dev::Drivers::yarpdev(int, char**): + Or append "--verbose" option to get more information.

**Comment by Nicogene** [comment, marked as answer]
Hi @YANGGANG1971 ,

`grabber` is just the server that publish the images over the yarp network.
Depending on the version of `yarp`/`icub-main` you have you have to enable the flag [`ENABLE_yarpmod_usbCamera`](https://github.com/robotology/yarp/blob/master/src/devices/usbCamera/CMakeLists.txt#L21) in YARP **OR** `ENABLE_icubmod_usbCamera` in `icub-main` while running `cmake`/`ccmake`.

Originally was in the `icub-main` repository, at some point it has been moved to `yarp`.

**Comment by YANGGANG1971** [comment]
My yarp/icub_main version is v1.15.0.
I did not find the options of ENABLE_usbCamera in YARP and ENABLE_icubmod_usbCamera in icub_main.
But, I found the ENABLE_yarpmod_usbCamera and ENABLE_yarpmod_usbCameraRaw in YARP as follows.
![Screenshot from 2020-06-29 23-42-39](https://user-images.githubusercontent.com/55526431/86020304-b7b6e700-ba62-11ea-8ddd-d1c79748597f.png)
But I ran into some errors as follows.
![Screenshot from 2020-06-29 23-41-16](https://user-images.githubusercontent.com/55526431/86020371-cb624d80-ba62-11ea-9e08-2da873c0cffd.png)

**Comment by Nicogene** [comment]
`ENABLE_yarpmod_usbCamera` is the correct option sorry, you have to install `video 4 linux` first:
```bash
sudo apt install libv4l-dev
```

**Comment by YANGGANG1971** [comment]
Okay, I will test it and let you know. Thanks a lot.

**Comment by pattacini** [comment]
Closing, feel free to open it up again if needed.
